lib/src/reporter.dart

```diff
@@ -93,7 +93,10 @@ class Reporter {
       color && message.isNotEmpty ? pen(message) : message;
 
   void _log(IOSink sink, String message, {bool shout: false}) {
-    if (quiet && !shout) return;
+    // This may look like a strange conditional, but please leave it.
+    // Somehow when using the tearoff `reporter.log` the `quiet` variable
+    // is null, which causes a runtime error when running tests in Dart 2.1.0
+    if (quiet == true && shout != true) return;
```

Comment:
Could it be that `Reporter` isn't getting passed both arguments during intiialization?
https://github.com/Workiva/dart_dev/blob/7eee6da5affe23b2f9cd1b8241a18431aafaff7c/lib/src/reporter.dart#L22

Based on the existing code, 
https://github.com/Workiva/dart_dev/blob/a3b36d0b7be50338751d421cf3c3813f4475f1c5/lib/src/reporter.dart#L30-L35
, I would expect `new Reporter()` to have both fields be null.

I feel like it should just be:
```dart
bool color;
bool quiet;

Reporter({bool this.color = true, bool this.quiet = false}); 
```

Could it be that it's only showing up in testing because this is being run in checked mode?